src/main/java/core/basesyntax/model/FruitTransaction.java

```diff
@@ -0,0 +1,52 @@
+package core.basesyntax.model;
+
+import java.util.Arrays;
+
+public class FruitTransaction {
+    private Operation operation;
+    private String fruit;
+    private int quantity;
+
+    public FruitTransaction(Operation operation, String fruit, int quantity) {
+        this.operation = operation;
+        this.fruit = fruit;
+        this.quantity = quantity;
+    }
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public String getFruit() {
+        return fruit;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public enum Operation {
+        BALANCE("b"),
+        SUPPLY("s"),
+        PURCHASE("p"),
+        RETURN("r");
+
+        private final String code;
+
+        Operation(String code) {
+            this.code = code;
+        }
+
+        public String getCode() {
+            return code;
+        }
+
+        public static Operation getOperationByCode(String code) {
+            return Arrays.stream(Operation.values())
+                    .filter(c -> c.getCode().equals(code))
+                    .findFirst()
+                    .orElseThrow(() -> new RuntimeException("Unknown code operation \""
+                            + code + "\""));
+        }
```

Comment:
In my opinion, it'd be better if you have this method within ParserServiceImpl.

Reply:
Let it be here, not critical.